wgpu-hal/src/gles/command.rs

```diff
@@ -148,8 +164,7 @@ impl super::CommandEncoder {
     fn set_pipeline_inner(&mut self, inner: &super::PipelineInner) {
         self.cmd_buffer.commands.push(C::SetProgram(inner.program));
 
-        //TODO: push constants
-        let _ = &inner.uniforms;
+        self.state.uniforms = inner.uniforms.clone();
```

Comment:
We really don't want to clone a heap-allocated container here.
Instead, it should do something like this:
```rust
self.state.uniforms.clear();
self.state.uniforms.extend_from_slice(&inner.uniforms);
```

Reply:
This is using a `Box` so it can't be done this way but I agree with you that cloning a `Box` isn't ideal

Reply:
Can't you switch to a `Vec<>`?